fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenePostingsTest.java

```diff
@@ -0,0 +1,414 @@
+                    1624L, "Document 2"));
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_POSTINGS_FIELD_METADATA).getCount() >= 3);
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_POSTINGS_TERM).getCount() >= 6);
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_POSTINGS_POSITIONS).getCount() >= 6);
+            assertTrue(timer.getCounter(LuceneEvents.Waits.WAIT_LUCENE_READ_POSTINGS_FIELD_METADATA).getCount() >= 3);
+            assertTrue(timer.getCounter(LuceneEvents.Waits.WAIT_LUCENE_READ_POSTINGS_TERMS).getCount() >= 3);
+            try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                final String[] strings = directory.listAll();
+                assertFieldCountPerSegment(directory, List.of("_0", "_1", "_2", "_3"), List.of(1, 1, 1, 0));
+            }
+        }
+    }
+
+    @Test
+    void testInsertDeleteDocuments() throws Exception {
+        Index index = SIMPLE_TEXT_SUFFIXES_WITH_OPT_POSTINGS;
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, 2.0)
+                .build();
+
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, "Document 1", 2));
+            recordStore.saveRecord(createSimpleDocument(1624L, "Document 2", 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, "NonDocument 3", 2));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.deleteRecord(Tuple.from(1623L));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.deleteRecord(Tuple.from(1547L));
+            context.commit();
+        }
+        try (FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            final RecordQuery query = buildQuery("Document", Collections.emptyList(), SIMPLE_DOC);
+            queryAndAssertFields(query, "text", Map.of(1624L, "Document 2"));
+            LuceneIndexTestUtils.mergeSegments(recordStore, index);
+            try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                // After a merge, all tombstones are removed and one document remains
+                assertFieldCountPerSegment(directory, List.of("_0", "_1", "_2"), List.of(0, 0, 1));
+            }
+            assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_DELETE_POSTINGS).getCount() > 0);
```

Comment:
This doesn't happen until after the merge, perhaps worth asserting that is the case.